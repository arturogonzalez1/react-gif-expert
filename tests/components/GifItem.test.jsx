import { fireEvent, render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';

describe('Tests in <GifItem />', () => { 

    const gif = {
        title: 'Testing',
        url: 'https://media2.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=a1b35471wn01jazq3si2xfrmhhgzm69swo4z05tuy35goyep&rid=giphy.gif&ct=g',
    }

    test('Should match with snapshot', () => { 
        
        const { container } = render(<GifItem { ...gif }/>);
        expect(container).toMatchSnapshot();

    });

    test('Should show an especific url and alt', () => { 

        render(<GifItem { ...gif }/>);

        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(gif.url);
        expect(alt).toBe(gif.title);

    });

    test('Should show title in the component', () => { 

        render(<GifItem { ...gif }/>);
        expect(screen.getByText(gif.title)).toBeFalsy();

    });

});