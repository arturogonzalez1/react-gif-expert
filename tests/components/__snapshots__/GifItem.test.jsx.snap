// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifItem /> Should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Testing"
      src="https://media2.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=a1b35471wn01jazq3si2xfrmhhgzm69swo4z05tuy35goyep&rid=giphy.gif&ct=g"
    />
    <p>
      Testing
    </p>
  </div>
</div>
`;
